keyring.php

```diff
@@ -392,7 +392,7 @@ static function get_validated_parameters( $parameters ) {
 		$return_hash = $parameters['hash'];
 		unset( $parameters['hash'] );
 
-		if ( self::get_parameter_hash( serialize( $parameters ) ) !== $return_hash ) {
+		if ( self::get_parameter_hash( json_encode( $parameters ) ) !== $return_hash ) {
```

Comment:
It's good practice to use `hash_equals()` for hash comparisons, instead of standard comparison operators. It probably doesn't matter so much in this particular instance, but it's better to have the habit of always using it.

Reply:
Done.